cavern/src/main/java/org/opencadc/cavern/files/CavernURLGenerator.java

```diff
@@ -146,28 +144,26 @@ public List<Protocol> getEndpoints(VOSURI target, Transfer transfer, Job job, Li
         if (transfer == null) {
             throw new IllegalArgumentException("transfer is required");
         }
-        List<Protocol> ret = null;
+        List<Protocol> ret;
         try {
-            Direction dir = transfer.getDirection();
-            PathResolver ps = new PathResolver(nodePersistence, authorizer, true);
-            Node n = ps.getNode(target.getParentURI().getPath());
-            // assume not null and Container already checked by caller (TransferRunner)
-            ContainerNode parent = (ContainerNode) n;
-            Node node = nodePersistence.get(parent, target.getName());
-
-            Subject currentSubject = AuthenticationUtil.getCurrentSubject();
-            if (Direction.pushToVoSpace.equals(dir) && node == null) {
-                // create new data node?? this currently does not happen because the library
-                // creates the DataNode
-                ret = handleDataNode(parent, target.getName(), transfer, currentSubject);
-            } else if (node instanceof DataNode) {
-                ret = handleDataNode(parent, target.getName(), transfer, currentSubject);
-            } else if (node instanceof ContainerNode) {
-                ret = handleContainerMount(target.getPath(), transfer, currentSubject);
+            PathResolver pr = new PathResolver(nodePersistence, authorizer, true);
+            PathResolver.Resolved resolved = pr.resolveNode(target);
```

Comment:
I thought the idea was that this should be able to use `Node node = pr.get(target path)` now? The spec requires that the DataNode is created during transfer negotiation. 

Admittedly, "transfer negotiation" spans multiple requests and this code (URL generation) happens in a later request but I think it was clear that we create the target DataNode in the first step (TransferRunner + PushToVOSpaceNegotiation) because we want to do all the checking in the first call rather than later... otoh, it is feasible that the new DataNode gets deleted by some other actor between when it is created in TransferRunner and then this code runs in TransferDetailsServlet, which could be an arbitrary amount of time later. So this code has to check that the DataNode still exists but should fail if it does not.

Given that we already noted that the DataNode create in PushToVOSpaceNegotiation must be consistent with whatever CreateNodeAction does, having another possible create in here is a bad idea.